packages/next/build/post-processing/serverless-build-id.ts

```diff
@@ -0,0 +1,40 @@
+import { readFile, writeFile } from 'fs'
+import MagicString from 'magic-string'
+import path from 'path'
+import { promisify } from 'util'
+
+import { recursiveReadDir } from '../../lib/recursive-readdir'
+
+const readFileP = promisify(readFile)
+const writeFileP = promisify(writeFile)
+
+const NEXT_REPLACE_BUILD_ID = '__NEXT_REPLACE__BUILD_ID__'
+
+export async function injectBuildId(distDir: string, buildId: string) {
```

Comment:
Can't we inject the buildId after Terser has ran through a webpack plugin? That way we don't have to do any filesystem operations.